dotnet/src/Microsoft.Agents.AI.VectorDataMemory/ChatHistoryMemoryProvider.cs

```diff
@@ -0,0 +1,453 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.VectorData;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Agents.AI.VectorDataMemory;
+
+/// <summary>
+/// A context provider that stores all chat history in a vector store and is able to
+/// retrieve related chat history later to augment the current conversation.
+/// </summary>
+/// <remarks>
+/// <para>
+/// This provider stores chat messages in a vector store and retrieves relevant previous messages
+/// to provide as context during agent invocations. It uses the VectorStore and VectorStoreCollection
+/// abstractions to work with any compatible vector store implementation.
+/// </para>
+/// <para>
+/// Messages are stored during the <see cref="InvokedAsync"/> method and retrieved during the
+/// <see cref="InvokingAsync"/> method using semantic similarity search.
+/// </para>
+/// <para>
+/// Behavior is configurable through <see cref="ChatHistoryMemoryProviderOptions"/>. When
+/// <see cref="ChatHistoryMemoryProviderOptions.SearchBehavior.OnDemandFunctionCalling"/> is selected the provider
+/// exposes a function tool that the model can invoke to retrieve relevant memories on demand instead of
+/// injecting them automatically on each invocation.
+/// </para>
+/// </remarks>
+public sealed class ChatHistoryMemoryProvider : AIContextProvider, IDisposable
```

Comment:
Can we also have samples added showing how this new components would be used?

This helps understanding in the bigger picture.